### PR TITLE
Preserve btrfs options for root (bsc#965279)

### PR DESCRIPTION
**Pull request by ancorgs**
It also includes some code cleanup and documentation that I needed to add to understand the code.

### CODE REVIEW

File: src/include/partitioning/custom_part_lib.rb [hunk 1]
```diff
@@ -414,54 +414,45 @@ def SubvolPart(can_do_subvol)
...
-          Ops.get_symbol(ret, "used_fs", :unknown)
-        )
+      ret = data.dup
+      if ret["mount"] == "/"
+        Builtins.y2milestone("before HandleSubvol fs:%1", ret['used_fs'])
         Builtins.y2milestone(
           "before HandleSubvol subvol:%1 userdata:%2",
-          Ops.get_list(ret, "subvol", []),
-          Ops.get_map(ret, "userdata", {})
+          ret['subvol'],
```
Comment (imobach) [inline src/include/partitioning/custom_part_lib.rb]: double quotes are preferred?

File: src/include/partitioning/custom_part_lib.rb [hunk 2]
```diff
@@ -672,7 +676,8 @@ def HandlePartWidgetChanges(init, ret, file_systems, old, new)
       end
 
       # set btrfs subvolumes (bnc#872210)
-      if init && new.fetch("mount", "") == "/"
+      # only when requested from the caller (i.e. new partition) (bsc#965279)
+      if init && new.fetch("mount", "") == "/" && force_defaults
```
Comment (imobach) [inline src/include/partitioning/custom_part_lib.rb]: `new`: what a dangerous name :) I know it was not introduced in this commit.

**Comment by imobach** [comment]
Just some minor nitpicking... otherwise, LGTM.

File: src/include/partitioning/custom_part_lib.rb (hunk 2 above)
Comment (ancorgs) [inline src/include/partitioning/custom_part_lib.rb]: With a smaller or more tested method, I would have changed it to `new_partition`, like I did in my previous PR. But I don't dare in this case.

Comment (imobach) [inline src/include/partitioning/custom_part_lib.rb]: Yes, I agree. I was pointing out that it's not the best name in Ruby :)

**Comment by imobach** [comment]
re-LGTM

**Comment by ancorgs** [comment]
Oh, it broke the old testsuite :-(

**Comment by imobach** [comment]
LGTM

**Comment by imobach** [comment]
LGTM. Let's see whether Travis is happy.

**Pull request merged**